src/access_control/rules.rs

```diff
@@ -58,11 +58,13 @@ pub enum Rule {
 
 /// JSON string representation prefix of Rule::StartsWith variant.
 const STARTS_WITH: &str = "starts-with:";
-const ANY: &str = "any";
+const ANY: &str = "*";
 
 impl From<String> for Rule {
     fn from(full_str: String) -> Self {
-        if let Some(s) = full_str.strip_prefix(STARTS_WITH) {
+        if full_str == ANY {
+            Rule::Any()
```

Comment:
This is the fix. The rest in this PR are just refactorings that I noticed while digging into this issue.